sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -102,6 +104,7 @@
      */
     CryptographyAsyncClient(String keyId, HttpPipeline pipeline, CryptographyServiceVersion version) {
         unpackAndValidateId(keyId);
+        this.keyId = keyId;
```

Comment:
Why doesn't this constructor `initializeCryptoClients()` while the other constructor does (line 95)?

Reply:
In other constructor, JWK key is passed into the constructor. Key is available to initialize crypto clients.
While in this constructor only key id is available at constructor time. In this scenario the Key gets fetched from the service and crypto clients are initialized on demand when user invokes a crypto operation on the client.